modules/finance/src/main/java/com/opengamma/strata/finance/rate/bond/FixedCouponBondPaymentPeriod.java

```diff
@@ -0,0 +1,762 @@
+/**
+ * Copyright (C) 2015 - present by OpenGamma Inc. and the OpenGamma group of companies
+ *
+ * Please see distribution for license.
+ */
+package com.opengamma.strata.finance.rate.bond;
+
+import static com.google.common.base.MoreObjects.firstNonNull;
+
+import java.io.Serializable;
+import java.time.LocalDate;
+import java.time.temporal.TemporalAdjuster;
+import java.util.Map;
+import java.util.NoSuchElementException;
+import java.util.Set;
+
+import org.joda.beans.Bean;
+import org.joda.beans.BeanDefinition;
+import org.joda.beans.ImmutableBean;
+import org.joda.beans.ImmutableConstructor;
+import org.joda.beans.JodaBeanUtils;
+import org.joda.beans.MetaProperty;
+import org.joda.beans.Property;
+import org.joda.beans.PropertyDefinition;
+import org.joda.beans.impl.direct.DirectFieldsBeanBuilder;
+import org.joda.beans.impl.direct.DirectMetaBean;
+import org.joda.beans.impl.direct.DirectMetaProperty;
+import org.joda.beans.impl.direct.DirectMetaPropertyMap;
+
+import com.google.common.collect.ImmutableSet;
+import com.opengamma.strata.basics.currency.Currency;
+import com.opengamma.strata.basics.index.Index;
+import com.opengamma.strata.collect.ArgChecker;
+import com.opengamma.strata.finance.rate.swap.PaymentPeriod;
+
+/**
+ * A period over which a fixed coupon is paid.
+ * <p>
+ * A single payment period within a fixed coupon bond, {@link ExpandedFixedCouponBond}.
+ * The payments of the fixed coupon bond consist periodic coupon payments and nominal payment.
+ * This class represents a single payment of the periodic payments. 
+ */
+@BeanDefinition
+public final class FixedCouponBondPaymentPeriod
+    implements PaymentPeriod, ImmutableBean, Serializable {
+
+  /**
+   * The primary currency of the payment period.
+   * <p>
+   * The amounts of the notional are usually expressed in terms of this currency,
+   * however they can be converted from amounts in a different currency.
+   */
+  @PropertyDefinition(validate = "notNull", overrideGet = true)
+  private final Currency currency;
+  /**
+   * The notional amount, positive if receiving, negative if paying.
+   * <p>
+   * The notional amount applicable during the period.
+   * The currency of the notional is specified by {@code currency}.
+   */
+  @PropertyDefinition
+  private final double notional;
```

Comment:
In the "FixedCouponBond", the notional is represented by a "CurrencyAmount notionalAmount;" while here it is represented by a Currency and separately a double for the notional. Why this difference of implementation?